### ISSUE TITLE
Unhandled date format passed to MockDate constructor: Fri, 26 Jul 2019 10:32:24 GMT

### ISSUE THREAD

**Issue by phawxby** [Jimbly/timezone-mock#24]
Works fine when passed into a normal Date object.

```
new Date('Fri, 26 Jul 2019 10:32:24 GMT')
Fri Jul 26 2019 11:32:24 GMT+0100 (British Summer Time)
```

**Comment by AleksandrukTad** [comment]
Steps to reproduce: 
`
    const test = new Date("2015-01-01T12:00:00").toUTCString();
    const isoTest = new Date(test).toISOString();
    returns: 2015-01-01T12:00:00.000Z

    register("Europe/London");
    const test = new Date("2015-01-01T12:00:00").toUTCString();
    const isoTest = new Date(test).toISOString();
    returns: Unhandled date format passed to MockDate constructor: Thu, 01 Jan 2015 12:00:00 GMT
`

**Comment by pavellishin** [comment]
Seconding this; `Joi` apparently depends on `Date` to perform date validation, so this non-matching behavior is breaking unrelated validation tests.

**Comment by ryanau** [comment]
Any update on this?

**Comment by Jimbly** [comment]
Pull requests to fix specific issues are welcome!  Shouldn't be too hard to support the inverse of toUTCString().  Supporting all date formats including locale-specific ones is impractical.  Generally returning "invalid date" for anything we don't know how to parse is a bad idea, though, that would make tests that might catch bugs silently start passing.  I'd be fine with an optional flag that does that behavior though - will make it hard to track down a test failure because of an obscure/unsupported Date format vs a test failure because of a bug, though.  I'd really just recommend `unregister`ing the mock timezone when testing code that is expected to be passing invalid strings to the Date constructor.  But, since the native Date constructor is fairly browser and locale-specific, it also seems like it's finding a "bug" already =).

**Comment by pavellishin** [comment]
> Generally returning "invalid date" for anything we don't know how to parse is a bad idea, though

I would imagine that acting as closely to the real `Date` constructor as possible would be a good thing, no?

**Comment by Jimbly** [comment]
> I would imagine that acting as closely to the real Date constructor as possible would be a good thing, no?

It is, but the real Date constructor handles parsing all sorts of date formats, including lots of non- or poorly-documented ones on various platforms, and mocking that is outside of the scope of this module.  Returning "Invalid Date" for something we don't know how to parse (that may be parsed just fine by a real Date constructor on some platforms) is *not* acting closely to the real Date constructor, and will cause confusing bugs in *other* code locations, instead of an exception here identifying the root cause.  If we could tell "this is definitely not something that is parsed by any version of Date" and return "Invalid Date", that'd be great, but that's basically "implement all possible date formats" =).

Anyway, definitely not something we'd want on by default, but totally fine for an opt-in thing, if you've got tests which you know are passing unparseable dates.

**Comment by apexearth** [comment]
I think to at least have an option allowing the disable of this protection would be super helpful.

`timezone_mock.register('UTC', { throwUnhandledFormats: false })`

In these circumstances, return with the `Invalid Date` `Date`.

**Comment by Jimbly** [comment]
Yeah, that sounds good to me.  PRs welcome =).

**Comment by apexearth** [comment]
How does this look?

**Comment by Jimbly** [comment]
@apexearth's change has been merged and published to NPM as v1.1.3 - for the secondary issue in this thread (passing a known invalid date format) you can work around it with the new `fallbackFn` option, however if you're passing a theoretically valid date format that simply isn't yet supported by this module, it'll just be potentially silently failing, so this doesn't help with the original issue here.
